src/main/java/hudson/model/ExternalJob.java

```diff
@@ -76,7 +78,11 @@ public synchronized ExternalRun newBuild() throws IOException {
         long timeSinceLast = System.currentTimeMillis() - lastBuildStartTime;
         if (timeSinceLast < 1000) {
             try {
-                Thread.sleep(1000 - timeSinceLast);
+                while(timeSinceLast > 0){
+                    wait();
```

Comment:
You can either keep the original code and suppress the warning (we intend to sleep inside the lock—to support the second-granularity timestamps we must); or update the baseline and delete the hack. My preference would be the latter.